service-base/src/main/java/org/eclipse/hono/service/credentials/validators/HashedPasswordValidator.java

```diff
@@ -0,0 +1,99 @@
+/**
+ * Copyright (c) 2017 Bosch Software Innovations GmbH.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    Bosch Software Innovations GmbH - initial creation
+ */
+package org.eclipse.hono.service.credentials.validators;
+
+import java.io.UnsupportedEncodingException;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Arrays;
+import java.util.Base64;
+import java.util.Map;
+
+import org.eclipse.hono.util.CredentialsConstants;
+import org.springframework.stereotype.Component;
+
+/**
+ * Validator to validate credentials of type {@link CredentialsConstants#SECRETS_TYPE_HASHED_PASSWORD} for a given password.
+ */
+@Component
+public final class HashedPasswordValidator extends AbstractSecretsValidator<String> {
+
+    /**
+     * Get the type of credentials secrets this validator is responsible for.
+     *
+     * @return The type of credentials secrets.
+     */
+    @Override
+    public String getSecretsType() {
+        return CredentialsConstants.SECRETS_TYPE_HASHED_PASSWORD;
+    }
+
+    /**
+     * Approve a single secret (as JsonObject) against a given password.
+     * If the password matches against it, the approval is considered successful.
+     * <p>This involves the hash-function found in the credentials secrets record to hash the given password before the
+     * comparison against the password in the credentials secrets record is done.
+     * <p>If a salt is contained in the credentials record (in base64 encoding), the hash function will be salted first.
+     *
+     * @param password The password to validate.
+     * @param aSecret The secret record as JsonObject (as returned by the <a href="https://www.eclipse.org/hono/api/Credentials-API/">Credentials API</a>.
+     * @return The result of the approval as boolean.
+     */
+    @Override
+    protected boolean validateSingleSecret(final String password, final Map<String, String> aSecret) {
```

Comment:
my point was that it is not an abstract structure representing any kind of secret but instead, it represents a hashed password (and its accompanying properties), because we are in the `HashedPasswordValidator` ...